src/libcore/sync/atomic.rs

```diff
@@ -941,8 +942,10 @@ impl<T> AtomicPtr<T> {
     ///
     /// let other_ptr   = &mut 10;
     /// let another_ptr = &mut 10;
-    ///
-    /// let value = some_ptr.compare_and_swap(other_ptr, another_ptr, Ordering::Relaxed);
+    /// // here `some_ptr` will not change as `other_ptr` has a different address from `ptr`
+    /// asserteq!(some_ptr.compare_and_swap(other_ptr, another_ptr, Ordering::Relaxed), 5);
```

Comment:
All this code is a little bit wrong (lacking dereferences where necessary). I.e. it asserts equality to `5`, but the function returns a `*mut T`, which cannot be dereferenced safely.